src/VolleyManagement.API/Startup.cs

```diff
@@ -0,0 +1,128 @@
+﻿using System;
+using Microsoft.AspNetCore.Authentication.Cookies;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Controllers;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using SimpleInjector;
+using SimpleInjector.Integration.AspNetCore.Mvc;
+using SimpleInjector.Integration.WebApi;
+using SimpleInjector.Lifestyles;
+using System.Threading.Tasks;
+using System.Web.Http;
+using VolleyManagement.API.Infrastructure;
+using VolleyManagement.Crosscutting.IOC;
+using VolleyManagement.Data.MsSql.Context;
+using VolleyManagement.Data.MsSql.Infrastructure;
+using VolleyManagement.Services.Infrastructure;
+
+namespace VolleyManagement.API
+{
+    public class Startup
+    {
+        private const string COOKIE_NAME = "VMCookie";
+
+        public Startup(IConfiguration configuration)
+        {
+            Configuration = configuration;
+            InitializeConnectionString();
+        }
+
+        public IConfiguration Configuration { get; }
+
+        // This method gets called by the runtime. Use this method to add services to the container.
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddCors();
+            services.AddMvc();
+
+            IntegrateSimpleInjector(services);
+
+            services.AddIdentity<IdentityUser, IdentityRole>();
+            services.AddAuthentication(CookieAuthenticationDefaults.AuthenticationScheme).
+                AddCookie(COOKIE_NAME, ConfigureCookies());
+
+            services.Configure<MvcOptions>(options => { options.Filters.Add(new RequireHttpsAttribute()); });
+        }
+
+        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
+        public void Configure(IApplicationBuilder app, IHostingEnvironment env)
+        {
+            app.UseCors(builder => builder.AllowAnyOrigin().AllowAnyMethod().AllowAnyHeader().AllowCredentials());
+
+            app.UseAuthentication();
+
+            if (env.IsDevelopment())
+            {
+                app.UseDeveloperExceptionPage();
+            }
+
+            app.UseMvcWithDefaultRoute();
```

Comment:
Do we need Mvc here?

Reply:
Or it is the same as web api in asp.net core?

Reply:
As far as I remember UseMvc was added on project creation.
It is possible that there was no need to change it to UseMvcWithDefaultRoute(adds MVC to the IApplicationBuilder request execution pipeline with a default route named 'default' and the following template: '{controller=Home}/{action=Index}/{id?}').

Reply:
There is no such method as UseWebApi. In asp.net core all controllers inherit from Microsoft.AspNetCore.Mvc.Controller